src/main.rs

```diff
@@ -118,6 +121,33 @@ fn run() -> i32 {
         },
     };
 
+    if matches.opt_present("s") {
+        let cursor = match display.get_cursor_position() {
+            Some(p) => p,
+            None => {
+                eprintln!("Failed to get cursor position");
+                return 1;
+            }
+        };
+
+        let screen_rects = match display.get_screen_rects(display.get_default_root()) {
```

Comment:
This introduces an additional call to `get_screen_rects` which is a little inefficient. It should be cached, or better yet lazily initialized so it only gets called when needed. [`std::Lazy`](https://doc.rust-lang.org/std/lazy/struct.Lazy.html) is currently experimental, so we'll need to find a crate for this.
Since `get_screen_rects` returns an iterator, you'll need to collect it into a vector, and then make the masking step reuse that. Maybe removing the iterator altogether would be a good idea. I believe I originally wrote it this way to avoid superfluous allocations, but that's already out of the window, so there's no point in keeping it.

We also already cache the result of `get_default_root` during initialization. You should definitely use that.

Reply:
As for this part, I think it should be fine if we cache it for now using a vector, I think we can still change that later on.

Reply:
Yes, do that, and adapt the masking step to reuse the same vector. Lazy evaluation might not even be necessary considering that masking always happens so forget about that.